### DISCUSSION TITLE
How do I register a custom marker with arguments

### DISCUSSION THREAD

**Discussion by FelixKJose** [pytest-dev/pytest#7897] [Q&A] [answered]
Hello All,
I have scenario where I have to mark my tests with JIRA requirements and then using `pytest_collection_modifyitems`  I have to read those custom marker arguments(requirement numbers) to a YML file. I am successfully able to do that. 

**But I am getting following warning when I run pytest:**

>  @pytest.mark.requirements(names="CARE-3206")
> Unknown pytest.mark.requirements - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
>     @pytest.mark.requirements(names=("CARE-3204", "CARE-3206"))

**Content of my pytest.ini:**
[pytest]
markers =
    requirements(names): mark test to show that the test is covering specified requirements

**My test case Sample:**
@pytest.mark.requirements(names="CARE-3206")
def test_version2_fail():
    assert ......

What should I do to avoid this warning?

**Comment by The-Compiler** [comment, marked as answer]
The config file just takes the name of a marker (not its signature), so you'd just use `requirements: mark test to show ...` there.

**Comment by FelixKJose** [comment]
@The-Compiler I have updated the pytest.ini as follows as you suggested:
```
[pytest]
markers =
    requirements: mark test to show that the test is covering specified requirements.
```

**I am still getting the above error if I just run:**

`pytest `

from the project directory

But I am not getting error when  I run:

`pytest tests/`

**Reply by The-Compiler** [reply]
I'm guessing your `pytest.ini` is in `tests/`? Either move it to the project root if you want to run pytest from there, or keep pytest configuration in a file in the project root (such as `pyproject.toml` / `setup.py` / `tox.ini`).

**Reply by FelixKJose** [reply]
Thank you @The-Compiler, yes it worked. Thank you so much
